server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -315,6 +316,31 @@ const orderMutations = {
           throw new ValidationFailed(`Only pending/expired orders can be marked as paid, this one is ${order.status}`);
         }
 
+        const hasUpdates = !isEmpty(difference(keys(args.order), ['id', 'legacyId']));
+        if (hasUpdates) {
```

Comment:
Not sure about that pattern. We're gonna send the values all the time from now on?

Should we check that it's set and different instead?

Anyway, that should work.

Reply:
No. This is backward compatible with existing API calls, and should only be used if necessary.

Reply:
With this code "hasUpdates" is misleading because it will tell you there is an update even if the amount has not changed. You will then write to the database while it's not needed.

It works but it's misleading and not necessarily optimal.

Reply:
Sure, I think it is not explicit enough. Implicitly it makes sense since none of these arguments are property of order and I'm not sure in which case someone would pass these amounts without consciously wanting to update them.

Reply:
> I'm not sure in which case someone would pass these amounts without consciously wanting to update them.

While Marking an order as paid without changing the values. That will be basically all the time?

Reply:
I thought that in such a case, the field would be empty and not pre-set with the existing value.
@SudharakaP can you confirm how exactly this works?

Reply:
I think always passing the value is prone to error. I think we should leave it empty so the user can know if they're updating the value or not.